harness/determined/deploy/local/cluster_utils.py

```diff
@@ -173,6 +181,18 @@ def master_up(
         master_conf["security"]["initial_user_password"] = initial_user_password
         make_temp_conf = True
 
+    try:
+        authentication.check_password_complexity(
+            master_conf["security"].get("initial_user_password")
+        )
+    except ValueError:
+        random_password_characters = string.ascii_uppercase + string.ascii_lowercase + string.digits
+        generated_user_password = "".join(
+            [secrets.choice(random_password_characters) for _ in range(16)]
```

Comment:
This isn't guaranteed to select a password which meets the security requirements. It could generate sixteen lowercase letters, for example.

The way I've done this in the past is to select one from each set, fill the rest of the length from the combined character set, then shuffle the list by swapping each element with a random value from 0-len.

That may not matter for local deployments, but I figured it wad worth mentioning.

Reply:
That's a good point -- I'll switch it to that. Thanks.